### PR TITLE
#4079 - Queue Monitoring - Schedulers Refactor (Part 7)

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Refactored the below schedulers and E2E tests.
- ece-process-integration
- ece-process-response-integration
  - _Note:_ some errors were converted to warnings because they are not recoverable and the job should not retry because those conditions would not be resolved.
- ier12-integration

### CODE REVIEW

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-request/ece-process-integration.scheduler.ts [hunk 1]
```diff
@@ -18,50 +21,35 @@ export class ECEProcessIntegrationScheduler extends BaseScheduler<void> {
...
+  ): Promise<string[] | string> {
+    const uploadResults = await this.eceFileService.processECEFile(
+      processSummary,
     );
-    this.logger.log("Executing ECE request file generation ...");
-    const uploadResults = await this.eceFileService.processECEFile();
-    this.logger.log("ECE request file generation completed.");
-
+    if (!uploadResults.length) {
+      return "No eligible COEs found.";
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-request/ece-process-integration.scheduler.ts:39]: 👍

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/ece-response-integration.scheduler.ts [hunk 2]
```diff
@@ -18,40 +21,34 @@ export class ECEResponseIntegrationScheduler extends BaseScheduler<void> {
...
     const processingResult = await this.eceResponseProcessingService.process();
-    this.logger.log("Processing ECE response files completed.");
-    return processingResult;
+    processingResult.forEach((result) => {
+      // Convert each ECE response processing result to a process summary for a location.
+      const locationResultSummary = new ProcessSummary();
+      processSummary.children(locationResultSummary);
+      result.summary.forEach((info) => locationResultSummary.info(info));
+      result.warnings.forEach((error) => locationResultSummary.warn(error));
+      result.errors.forEach((error) => locationResultSummary.error(error));
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/ece-response-integration.scheduler.ts:41]: Is there a particular reason why we did not use the process summary inside the processing service here?

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts [hunk 3]
```diff
@@ -19,53 +22,36 @@ export class IER12IntegrationScheduler extends BaseScheduler<GeneratedDateQueueI
...
-  async processIER12File(
-    job: Job<GeneratedDateQueueInDTO | undefined>,
-  ): Promise<QueueProcessSummaryResult[]> {
-    this.logger.log(
-      `Processing IER integration job ${job.id} of type ${job.name}.`,
-    );
-    this.logger.log("Executing IER 12 file generation ...");
+  protected async process(
+    job: Job<GeneratedDateQueueInDTO>,
+    processSummary: ProcessSummary,
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts:34]: Please add the comment for processSummary.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts [hunk 4]
```diff
@@ -19,53 +22,36 @@ export class IER12IntegrationScheduler extends BaseScheduler<GeneratedDateQueueI
...
-   * To be implemented.
-   */
-  protected async process(): Promise<string | string[]> {
-    throw new Error("Method not implemented.");
-  }
-
   /**
    * Identifies all the applications which are in assessment
    * for a particular institution and generate the request file.
    * @params job has generatedDate Date in which the assessment for
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts:28]: Minor and not part of the PR. The comment must be `@param`

File: sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts [hunk 5]
```diff
@@ -84,19 +87,21 @@ export class IER12ProcessingService {
...
       }
-    } catch (error) {
-      this.logger.error(`Error while uploading content for IER 12: ${error}`);
-      // TODO: On error, the error message must added to the upload result and
-      // processing must continue for the next institution without aborting.
-      throw error;
+    } catch (error: unknown) {
+      const errorMessage = "Error while uploading content for IER 12.";
+      this.logger.error(errorMessage, error);
+      processSummary.error(errorMessage, error);
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts:104]: What could be the benefit of not throwing the error here?

**Comment by dheepak-aot** [comment]
Good Job @andrewsignori-aot. Only minor comments.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/ece-response-integration.scheduler.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/ece-response-integration.scheduler.ts:41]: The service is heavily relying in the current summary and replacing it by the `ProcessSummary` would demand way more refactor, which I considered completely outside the target of this effort.

File: sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts:104]: First, the previous log that stated "processing must continue for the next institution without aborting." was misleading because this block is the last one before wrapping up the operation.
Throwing or not throwing here will not change the fact that the job will retry, but not throwing will allow the `IER12UploadResult` to be returned and logged before the job fails.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts:35]: Added.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts]: Changed.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/ece-response-integration.scheduler.ts (hunk 2 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/ece-response-integration.scheduler.ts:41]: Makes sense. Can we have a ticket created to not lose the track of that effort?

**Review by dheepak-aot — APPROVED** [review]
Thanks for making the changes. Looks good. 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot
